### DISCUSSION TITLE
[Compendium] Ideas to build with Semaphore

### DISCUSSION THREAD

**Discussion by andyguzmaneth** [semaphore-protocol/semaphore#463] [General]
Explore a list of ideas and cool projects to build with Semaphore. Even if some iterations of these ideas have already been built, you can take inspiration from them to build something new. If any of these ideas are interesting to you, comment on the discussion or join us on [Telegram](https://semaphore.pse.dev/telegram) and we can expand on a possible implementation with you!

- [ ] Account Abstraction
    - [x] ERC-4337 Compliant Wallet ([Semaphore Wallet](https://ethglobal.com/showcase/semaphore-wallet-s7ja9)).
    - [ ] Account Recovery.
    - [ ] Anon Multisig.
- [ ] Farcaster
    - [ ] Anonymous Polls and Voting.
    - [ ] Anonymous Reputation System.
- [ ] Governance
    - [ ] Anonymous Voting.
    - [ ] Anonymous Feedback.
- [x] Communications
    - [x] Anonymous Social Media ([ZK3](https://github.com/monemetrics) on Lens).
    - [x] Anonymous Chats ([Discreetly](https://pse.dev/projects/discreetly)).
    - [ ] Build a Telegram mini app with Semaphore.
- [x] Education - Private credentials and certificates ([Block Qualified](https://hackmd.io/@deenz/r1Tt-uFcj))
- [ ] Government
    - [ ] Privately prove income or residence to access government benefits and services.
    - [ ] Privately prove the number of countries one can visit with a certain passport.
- [ ] Healthcare
    - [ ] A system that allows patients to obtain prescriptions without revealing their identity.
    - [ ] A platform where patients can share their health data with researchers or healthcare providers anonymously.
    - [ ] A system where patients can prove their eligibility for health insurance or specific medical services without revealing their full identity.
- [ ] Gamification - A game that requires players to demonstrate their skills or resources without showing them directly.
- [ ] Build a private “[Secret Leader Election](https://ethereum.org/en/roadmap/secret-leader-election/)” protocol for validators.
- [ ] MPC - Simple Semaphore 3PC.
- [ ] W3C Standards - Integrate Semaphore with DIDs and VCs (e.g. [TW-DID](https://github.com/tw-did/tw-did)).
- [ ] Anonymity Sets - combine Coinbase authentication and Semaphore ID pool. ([idea thread](https://x.com/KamesGeraghty/status/1744787072836112667?s=20)).
- [x] DeFi - Anonymous DEX ([AX](https://github.com/hcheng826/anon-exchange-v2)).
- [ ] Metamask Snap to manage Semaphore identities and generate proofs.

We recommend focusing on identifying existing open-source projects in a specific area and enhancing them with Semaphore rather than building a complete solution from scratch. For example, if you want to build an app to share anonymous feedback, you could enhance existing public platforms like Decidim (an open-source participatory democracy tool) by integrating Semaphore as an extension instead of re-implementing an entire infrastructure.

**Comment by Michael679089** [comment]
>  Gamification - A game that requires players to demonstrate their skills or resources without showing them directly.

huh
